### DISCUSSION TITLE
Where should Wazuh Agent be run when using ECS Fargate?

### DISCUSSION THREAD

**Discussion by DylanVann** [wazuh/wazuh#15222] [Q&A] [answered]
I'm using AWS ECS to run a Next.js application. I'm currently running the Wazuh Agent in these Docker containers, but it makes them larger and slower to deploy. I would prefer to use Distroless as the base image for these containers, and just run the Node.js process.

I'm using Fargate to run the containers, so there's no EC2 host to install Wazuh Agent on. What would you recommend in this situation?

This comment suggests running Wazuh Agent in the Docker containers might not be the right way to go: https://github.com/wazuh/wazuh-packages/issues/451#issuecomment-668414067

**Comment by carlosdams** [comment, marked as answer]
Hi @DylanVann, 
You can consider [using the awslogs log driver](https://docs.aws.amazon.com/AmazonECS/latest/userguide/using_awslogs.html) and send the logs to CloudWatch, then you will use Wazuh to monitor the logs from CloudWatch.

The first step is creating the Log group which you can create from CloudWatch or allow Amazon ECS to auto-configure your CloudWatch logs, in the task definition you will add a logConfiguration.
Here is another link that explains a little bit easier the process of using the awslogs log driver: https://aws.amazon.com/blogs/compute/centralized-container-logs-with-amazon-ecs-and-amazon-cloudwatch-logs/

Once you are generating the logs or when your log group is created, you can follow the guide from Wazuh to monitor these logs: https://documentation.wazuh.com/current/amazon/services/supported-services/cloudwatchlogs.html#aws-cloudwatchlogs

I hope this is a good option for you, let me know

**Reply by DylanVann** [reply]
Hi @carlosdams thank you for the information!

So you'd recommend monitoring the logs over running the Wazuh Agent inside a Docker container on Fargate? Would you say the Wazuh Agent is more intended for container hosts or VMs?

We are able to run Wazuh Agent in Fargate, it's just a matter of if that's a good choice or not.

**Reply by carlosdams** [reply]
Both has it pros and cons, however, having the Agent running inside a container with other applications goes against the container model where each container should only fulfill one function. 
Some differences on what can be monitored can be found here: https://documentation.wazuh.com/current/getting-started/use-cases/container-security.html
The logs coming using the [awslogs log driver](https://docs.aws.amazon.com/AmazonECS/latest/userguide/using_awslogs.html) are similar to the ones of `STDOUT` and `STDERR` I/O streams from a terminal, if your application is logging to `/dev/stdout` and `/dev/stderr` as explained [here](https://docs.docker.com/config/containers/logging/) you will be getting very useful logs.

I personally have not tested running the Wazuh Agent in a container of ECS Fargate, would you tell me the high level steps you performed to achieve it? I'll appreciate it.
The resource consumption of a Wazuh Agent is minimal, regardless of whether it is containerized or a classical installation. Probably you can play with the configuration of your application and the Wazuh Agent to test if it improves the performance.

**Reply by DylanVann** [reply]
I'm more on the application side of things. I want the docker image to be minimal and deploy quickly. In this case having Wazuh agent running in the container complicates things. We're using a very large base image `phusion/baseimage`, doing a lot of complicated setup, running multiple processes, we have many detected potential vulnerabilities. I want to use Distroless instead but haven't been able to convince other developers that this isn't the best place to run the Wazuh agent.

Having you confirm that running the Wazuh agent inside the container "goes against the container model" and isn't highly recommended might help convince other devs I'm working with to look into alternatives, like monitoring the logs instead.

Thank you for the help!
